controlplane/pkg/model/model.go

```diff
@@ -20,13 +21,24 @@ type Dataplane struct {
 	RemoteMechanisms []*remote.Mechanism
 }
 
+type ClientConnection struct {
```

Comment:
It might be a good idea to store which 'side' of the Xcon the connection is on, which would allow retrieval of the complete Connection object.